_posts/en/newsletters/2019-03-19-newsletter.md

```diff
@@ -0,0 +1,277 @@
+  loss of funds.  Noinput can enable an alternative onchain enforcement
+  layer for LN that's more suitable for channels with multiple
+  participants, ultimately allowing a greater number of channels to be
+  opened in the same amount of block space.  Towns describes a
+  reasonably plausible worst-case scenario where failure to ensure
+  noinput safety puts adoption of other valuable protocol features at
+  risk.  To avoid that happening, he proposes refinements to earlier
+  ideas about output tagging (see [Newsletter #34][]) and a new
+  alternative that would require every transaction with a signature that
+  uses noinput to also contain a signature that doesn't use noinput.
+  This would prevent third-parties from executing replay attacks with
+  noinput transactions, but it would come with the downside of
+  moderately larger transaction sizes when noinput was being used.
+
+- **Esplora updated:** Nadav Ivgi [announced][ivgi twitter] an update of
+  this open source block chain explorer.  (See our coverage of its
+  initial release in [Newsletter #25][].)  Unconfirmed transactions are
+  now displayed with an estimate of how long it will take them to
+  confirm given current network conditions, or an overpayment warning if
+  they pay more than necessary to be confirmed quickly.  Perhaps more
+  notably, the details view for both confirmed and unconfirmed
+  transactions provides an analysis of the features and anti-features
+  used or omitted from the transaction.  For example: use of segwit,
+  address reuse, inconsistent output precision, matching an [unnecessary
+  input heuristic][uih2], inconsistent input/output script types, and
+  changeless transactions.
+
+    After seeing the new changes, Ryan Havar raised concerns [on
+    Reddit][havar reddit] about the possibly high rate of false positive
+    privacy warnings, leading to him opening an [issue][havar github] on
+    Esplora's GitHub about the problem.  Attempting to address
+    these concerns, Ivgi started a [conversation][esplora convo] with
+    several Bitcoin Core developers.  Privacy advocates may wish to
+    review this conversation, which covered topics such as:
+
+    - Gregory Maxwell and Pieter Wuille believe Bitcoin Core would
+      occasionally match Unnecessary Input Heuristic #2 (UIH2) since
+      the release of Bitcoin 0.1 in 2009, with increasing frequency in
+      more recent releases, making this heuristic less useful than
+      hypothesized for distinguishing between commercial services and
+      end-user wallets.
+
+    - Bitcoin Core's coin selection updates over the past two releases
+      allow it to produce transactions without change outputs quite
+      often.  These transactions are more efficient and more privacy
+      preserving than transactions with change, but Esplora currently
+      marks them in red as privacy-leaking because they also match the
+      pattern of sweep transactions.
```

Comment:
Suggesting to link sweep transactions for more information. I didn’t find a great definition on stack exchange or wiki. Perhaps to https://en.bitcoin.it/wiki/Privacy#Exact_payment_amounts ?